src/charm.py

```diff
@@ -446,7 +446,11 @@ def _reload_configuration(self) -> None:
         if self._is_config_valid() and container.can_connect():
             layer_config = self._create_layer_config()
             container.add_layer(SERVICE_NAME, layer_config, combine=True)
-            container.pebble.replan_services()
+            # Currently, pebble replan will cause
+            # ChangeError("cannot start service while terminating".)
+            # Link to issue: https://github.com/canonical/pebble/issues/206
+            container.stop(SERVICE_NAME)
+            container.start(SERVICE_NAME)
```

Comment:
This seems like we're going to be doing a stop and a start any time `_reload_configuration` is called, which might happen on any `config-changed`. I wonder if it's possible to only do a stop/start if we have recompiled the assets? Or do we think a stop/start is definitely needed any time `config-changed` is called?

Reply:
That is a good point! Would limiting the scope of this PR to the simple fix for the moment and creating the refactor as another issue work for you? Since container.replan() also internally stop/starts the services the effect is currently the same. From my limited knowledge, I think it may require a bit more planning to separate out the flows in which the stop/start service is definitely required.
What do you think? :)

Reply:
Yes, I think it'd be reasonable to do this in a separate PR as long as we get to it soon (before we begin production rollouts of this charm.

It might not be too complicated, I think. As I understand it pebble will restart a service if any environment variable you're passing in has changed (this will need testing). We could simply create a `DISCOURSE_ASSETS_LAST_UPDATED` environment variable with a string rendering of timestamp, for example, and update that any time we want the service to be restarted, and then continue with `container.replan_services`.

Anyway, feel free to close out this conversation and follow up in a separate PR.